### PR TITLE
NATIVE-104: Fix push reducer by adding the children to the models

### PR DESCRIPTION
**Pull request by maxammann**

### CODE REVIEW

File: src/modules/endpoint/reducers/pushCategory.js [hunk 1]
```diff
@@ -36,6 +37,7 @@ const pushCategory = (
 
   forEachTreeNode(root, node => categoriesMap.getChildren(node), depth, (node, children) => {
     resultModels[node.path] = node
+    Object.assign(resultModels, keyBy(children, child => child.path))
```
Comment (maxammann) [inline src/modules/endpoint/reducers/pushCategory.js]: I'm a bit unsure why we need to add the children. I mean they should be added by `resultModels[node.path] = node`

Comment (NKlug) [reply src/modules/endpoint/reducers/pushCategory.js]: I agree. I think all this line does is add 1 to the `depth`. This seems a bit odd to me.

File: src/modules/endpoint/reducers/pushCategory.js [hunk 2]
```diff
@@ -3,6 +3,7 @@
 import type { CategoriesStateType } from '../../app/StateType'
 import type { PushCategoryActionType } from '../../app/StoreActionType'
 import { CategoryModel } from '@integreat-app/integreat-api-client'
+import { keyBy } from 'lodash/collection'
 
 const forEachTreeNode = (
```
Comment (Schedulaar) [inline src/modules/endpoint/reducers/pushCategory.js]:
```js
const forEachTreeNode = (
  root: CategoryModel,
  resolveChildren: CategoryModel => Array<CategoryModel>,
  depth: number,
  nodeAction: (CategoryModel, Array<CategoryModel>) => void
) => {
  if (depth == 0) {
    nodeAction(root, null)
  } else {
    const children = resolveChildren(root)
    nodeAction(root, children)
    children.forEach(child => forEachTreeNode(child, resolveChildren, depth - 1, nodeAction))
  }
}
```

File: src/modules/endpoint/reducers/pushCategory.js [hunk 3]
```diff
@@ -36,6 +37,7 @@ const pushCategory = (
 
   forEachTreeNode(root, node => categoriesMap.getChildren(node), depth, (node, children) => {
     resultModels[node.path] = node
+    Object.assign(resultModels, keyBy(children, child => child.path))
     resultChildren[node.path] = children.map(child => child.path)
   })
```
Comment (Schedulaar) [inline src/modules/endpoint/reducers/pushCategory.js]:
```js
forEachTreeNode(root, node => categoriesMap.getChildren(node), depth, (node, children) => {
    resultModels[node.path] = node
    if (children !== null) {
      resultChildren[node.path] = children.map(child => child.path)
    }
  })
```

File: src/modules/endpoint/reducers/pushCategory.js [hunk 4]
```diff
@@ -4,15 +4,27 @@ import type { CategoriesStateType } from '../../app/StateType'
 import type { PushCategoryActionType } from '../../app/StoreActionType'
 import { CategoryModel } from '@integreat-app/integreat-api-client'
 
+/**
+ * Iterates through a tree until depth is reached. A depth of 0 means only the root is visited.
+ * A depth of 1 means the root and the children of it are visited. Please note that for each children of the root
+ * nodeAction is called once with the category and null for the children as parameters.
```
Comment (Schedulaar) [inline src/modules/endpoint/reducers/pushCategory.js]: Please note that for each node with depth `depth` null is passed as argument for `children` in nodeAction.

**Pull request merged**